site/blog/unlisted/subqueries.md

```diff
@@ -0,0 +1,86 @@
+---
+slug: subquery-vs-join
+title: Subqueries vs Joins
+authors: [depombo]
+tags: [sql]
+---
+
+SQL is a powerful language that is widely used in database management systems. It allows for the manipulation of large data sets by performing operations such as querying, inserting, updating, and deleting data. Two common techniques used in SQL for retrieving data from multiple tables are joins and subqueries. In this blog post, we will explore the differences between these two techniques and when it makes sense to use one over the other.
+
+### Refresher on joins and subqueries
+
+Joins are used to combine rows from two or more tables based on a related column between them. There are different types of joins in SQL, including inner join, left join, right join, and full outer join. Inner join retrieves only the rows that have matching values in both tables, while left join retrieves all the rows from the left table and the matching rows from the right table. Right join, on the other hand, retrieves all the rows from the right table and the matching rows from the left table. Full outer join retrieves all the rows from both tables, whether there are matching values or not.
+
+Subqueries are used to retrieve data from one table based on the results of a query on another table. They are enclosed in parentheses and can be used in different parts of a SQL statement, such as the `WHERE` clause, `HAVING` clause, and `FROM` clause. Subqueries can be correlated or non-correlated. Correlated subqueries are executed for each row of the outer query, while non-correlated subqueries are executed only once before the outer query is executed.
+
+### Know when the hammer suits the nail
+
+Joins and subqueries have their strengths and weaknesses. Joins are useful when retrieving data from multiple tables with related data, performing aggregation functions on related data, and when there are a small number of tables involved in the query. On the other hand, subqueries are useful when retrieving data from a table based on the results of a query on another table, performing complex calculations or conditional statements based on data from another table, and when there are multiple tables involved in the query and joins may become complex and difficult to manage.
```

Comment:
this paragraph seems a bit off to me. Firsts I think we are repeating too closely `related data`. And at the end I did not get if for complex queries is better subqueries or joins?